test/unit/errors/error-collector.test.js

```diff
@@ -275,128 +284,131 @@ describe('Errors', function () {
+      t.equal(errorTraces.length, 0)
 
       errorCollector.addUserError(null, error)
 
-      expect(errorTraces.length).equal(0)
+      t.equal(errorTraces.length, 0)
 
       agent.config.error_collector.enabled = true
+      t.end()
     })
 
-    it('should not gather errors if it is switched off by server config', function () {
+    t.test('should not gather errors if it is switched off by server config', (t) => {
       const error = new Error('this error will never be seen')
       agent.config.collect_errors = false
 
       const errorTraces = getErrorTraces(errorCollector)
-      expect(errorTraces.length).equal(0)
+      t.equal(errorTraces.length, 0)
 
       errorCollector.add(null, error)
 
-      expect(errorTraces.length).equal(0)
+      t.equal(errorTraces.length, 0)
 
       agent.config.collect_errors = true
+      t.end()
     })
 
-    it('should gather the same error in two transactions', function () {
+    t.test('should gather the same error in two transactions', (t) => {
       const error = new Error('this happened once')
       const first = new Transaction(agent)
       const second = new Transaction(agent)
 
       const errorTraces = getErrorTraces(agent.errors)
-      expect(errorTraces.length).equal(0)
+      t.equal(errorTraces.length, 0)
 
       agent.errors.add(first, error)
-      expect(first.exceptions.length).equal(1)
+      t.equal(first.exceptions.length, 1)
 
       agent.errors.add(second, error)
-      expect(second.exceptions.length).equal(1)
+      t.equal(second.exceptions.length, 1)
 
       first.end()
-      expect(errorTraces.length).equal(1)
+      t.equal(errorTraces.length, 1)
 
       second.end()
-      expect(errorTraces.length).equal(2)
+      t.equal(errorTraces.length, 2)
+      t.end()
     })
 
-    it('should not gather the same error twice in the same transaction', function () {
+    t.test('should not gather the same error twice in the same transaction', (t) => {
       const error = new Error('this happened once')
 
       const errorTraces = getErrorTraces(errorCollector)
-      expect(errorTraces.length).equal(0)
+      t.equal(errorTraces.length, 0)
 
       errorCollector.add(null, error)
       errorCollector.add(null, error)
-      expect(errorTraces.length).equal(1)
+      t.equal(errorTraces.length, 1)
+      t.end()
     })
 
-    it('should not break on read only objects', function () {
+    t.test('should not break on read only objects', (t) => {
       const error = new Error('this happened once')
       Object.freeze(error)
 
       const errorTraces = getErrorTraces(errorCollector)
-      expect(errorTraces.length).equal(0)
+      t.equal(errorTraces.length, 0)
 
       errorCollector.add(null, error)
       errorCollector.add(null, error)
 
-      expect(errorTraces.length).equal(1)
+      t.equal(errorTraces.length, 1)
+      t.end()
     })
 
-    describe('add()', function () {
-      let aggregator
+    t.test('add()', (t) => {
+      t.doesNotThrow(() => {
```

Comment:
nice. the old test is a big no no to me. running a test and relying on the fact it doesn't error is a bad practice

Reply:
Yeah, if I had been a little more careful I would have also converted a bunch of the single-assert tests from mocha into one big test with many asserts where each assert described what it's doing in tap, but I got a bit tired about the process.